Networking/Networking/Model/WCAnalyticsCustomer.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+public struct WCAnalyticsCustomer: Codable {
+
+    /// Unique identifier for the user
+    public let userID: Int64
+
+    /// Customer name
+    public let name: String?
+
+    /// WCAnalyticsCustomer struct Initializer
+    ///
+    public init(userID: Int64, name: String?) {
+        self.userID = userID
+        self.name = name
+    }
+
+    /// Public initializer for WCAnalyticsCustomer
+    ///
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let userID = try container.decode(Int64.self, forKey: .userID)
+        let name = try container.decode(String.self, forKey: .name)
+
+        self.init(userID: userID, name: name)
+    }
+}
+
+extension WCAnalyticsCustomer {
+    enum CodingKeys: String, CodingKey {
+        case userID =   "id"
+        case name   =   "username"
```

Comment:
Have you checked these against the Android implementation?

Reply:
Yeah! Android did not follow this approach initially, and it was decided to launch the feature as-it-is ( p1656431752334189-slack-CGPNUU63E ) and iterate from there based on feedback: Either leave it as it is,  implementing the proposed solution we're doing now on iOS ( pe5pgL-3r-p2 ), or using the API directly if a fix for searching first and last names on the `wc/v3/customers` endpoint came up ( it hasn't at the moment ).

So in this case I'm going a bit blind on this regard, but I asked for current implementation details ( p1664485668537849-slack-CGPNUU63E ) so I could follow their approach as well if needed. At this point I believe I will only need the `userID` in this case, but left the `username` for the moment so I could play with different datatypes if needed.

Reply:
Adding to this: Just wondering if would be helpful to add some extra fields here that (despite being optional) could be helpful to store the initial information for each customers, and not only their customer ID and username. For example this endpoint does not separate between firstName and lastName, as we want it to appear in the end, however there are other fields that do coincide, for example: Country, City, or PostCode. 

On one end I understand that we're already performing the request to fetch this data we could fill all possible fields in one go, however on other end, it may cause the problems we spoke in our last pairing when attempting to map and decode the data, as we could find unexpected input.

For now I'm inclined to use as little data as needed for this step (that's basically the customerID) and add the rest directly on the next steps when we actually fetch data from the specific `/wc/v3/customers/<id>` endpoint, which contains all we need. Thoughts?

Reply:
> For now I'm inclined to use as little data as needed for this step (that's basically the customerID) and add the rest directly on the next steps when we actually fetch data from the specific /wc/v3/customers/<id> endpoint, which contains all we need. Thoughts?

That's fine with me. You can save yourself some hassle if you get the overall set of properties correct when you do the Storage layer, so that you can make a single change to the Core Data model, but iterating on the Networking layer later is not expensive.